src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala

```diff
@@ -1,10 +1,31 @@
 package is.hail.expr.ir.functions
 
 import is.hail.expr.ir._
+import is.hail.expr.types
 import is.hail.expr.types._
 
 object DictFunctions extends RegistryFunctions {
 
+  def contains(dict: IR, key: IR) =
+    If(IsNA(dict),
+      NA(TBoolean()),
+      ApplyComparisonOp(
+        EQWithNA(key.typ),
```

Comment:
What is the semantics of `EQWithNA`?

Reply:
`EQ()` returns `NA()` if either object is missing; `EQWithNA()` never returns `NA()` and considers missing objects to be equvalent to each other but not to non-missing elements.